javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/SitewiseScenario.java

```diff
@@ -0,0 +1,438 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.example.iotsitewise.scenario;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.iotsitewise.model.CreateAssetModelResponse;
+import software.amazon.awssdk.services.iotsitewise.model.CreateAssetResponse;
+import software.amazon.awssdk.services.iotsitewise.model.IoTSiteWiseException;
+import software.amazon.awssdk.services.iotsitewise.model.ResourceAlreadyExistsException;
+import software.amazon.awssdk.services.iotsitewise.model.ResourceNotFoundException;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.concurrent.CompletableFuture;
+
+// snippet-start:[iotsitewise.java2.scenario.main]
+public class SitewiseScenario {
+
+    public static final String DASHES = new String(new char[80]).replace("\0", "-");
+
+    private static final Logger logger = LoggerFactory.getLogger(SitewiseScenario.class);
+    static Scanner scanner = new Scanner(System.in);
+
+    private static final String ROLES_STACK = "RoleSitewise";
+
+    static SitewiseActions sitewiseActions = new SitewiseActions();
+
+    public static void main(String[] args) throws Throwable {
+        final String usage = """
+            Usage:
+               <assetModelName> <assetName> <portalName> <contactEmail> <gatewayName> <myThing>
+
+            Where:
+                assetModelName - The name of the asset model used in the IoT SiteWise program.
+                assetName -  The name of the asset created in the IoT SiteWise program. 
+                portalName - The name of the IoT SiteWise portal where the asset and other resources are created.
+                contactEmail - The email address of the contact person associated with the IoT SiteWise program.
+                gatewayName - The name of the IoT SiteWise gateway used to collect and send data to the IoT SiteWise service.
+                myThing - The name of the IoT thing or device that is connected to the IoT SiteWise gateway.
+            """;
+
```

Comment:
Why does the user have to do this prep work? Just hard code the names like you did earlier.

Reply:
On;y email is an args now. Rest can use a default value.